### PR TITLE
actix-multipart: Correctly parse multipart body which does not end in CRLF

### PR DESCRIPTION
**Pull request by jefshe**
relates to #1038 
my reading of https://tools.ietf.org/html/rfc2046 AND https://tools.ietf.org/html/rfc7578 suggest to me that the server implementation should handle the case where no CRLF is present at the end of the multipart form. Here are the supporting paragraphs:

 ```
4.1.  "Boundary" Parameter of multipart/form-data

   As with other multipart types, the parts are delimited with a
   boundary delimiter, constructed using CRLF, "--", and the value of
   the "boundary" parameter.  The boundary is supplied as a "boundary"
   parameter to the multipart/form-data type.  As noted in Section 5.1
   of [RFC2046], the boundary delimiter MUST NOT appear inside any of
   the encapsulated parts, and it is often necessary to enclose the
   "boundary" parameter values in quotes in the Content-Type header
   field.
```
A "boundary" is considered to start with CRLF

```
     delimiter := CRLF dash-boundary

     close-delimiter := delimiter "--"

     preamble := discard-text

     epilogue := discard-text
```

```

     multipart-body := [preamble CRLF]
                       dash-boundary transport-padding CRLF
                       body-part *encapsulation
                       close-delimiter transport-padding
                       [CRLF epilogue]
```
The ending CRLF is considered part of the epilogue which can be ignored

### CODE REVIEW

**Review by fafhrd91 — CHANGES_REQUESTED** [review]
Could you add simple unit test and update changes.md

**Review by neoeinstein — CHANGES_REQUESTED** [review]
Could you add another test that doesn't include the final `CRLF` in the `tests` module?

File: actix-multipart/src/server.rs [hunk 1]
```diff
@@ -768,7 +767,7 @@ impl PayloadBuffer {
             .map(|idx| self.buf.split_to(idx + line.len()).freeze());
 
         if res.is_none() && self.eof {
-            Err(MultipartError::Incomplete)
+            Ok(self.buf.take().freeze())
```
Comment (JohnTitor) [inline actix-multipart/src/server.rs]: Mismatched types error here, probably `Ok(Some(self.buf.take().freeze()))`.

Comment (fafhrd91) [reply actix-multipart/src/server.rs]: i dont think this is right code. this could lead to partial reads reported as valid

Comment (jefshe) [reply actix-multipart/src/server.rs]: Yep good call, ive moved this out into a new function `readline_or_eof` which is called by `read_boundary`, a partial read should then result in a `BoundaryError`

**Comment by jefshe** [comment]
> 
> 
> Could you add another test that doesn't include the final `CRLF` in the `tests` module?

> 
> 
> Could you add simple unit test and update changes.md

Added a unit test now

**Review by leizzer — APPROVED** [review]

**Comment by fafhrd91** [comment]
thanks!

**Pull request merged**